docs/reference/performance.rst

```diff
@@ -0,0 +1,277 @@
+device memory into shared memory. This involves each thread of a block loading 
+data from device memory to shared memory, synchronizing with all other threads 
+of the block, processing the data in shared memory, synchronizing again if 
+necessary, and writing the results back to device memory.
+
+For some applications, a traditional hardware-managed cache is more appropriate
+to exploit data locality. On devices of certain compute capabilities, the same 
+on-chip memory is used for both L1 and shared memory, and the amount dedicated 
+to each is configurable for each kernel call.
+
+Finally, the throughput of memory accesses by a kernel can vary significantly 
+depending on the access pattern for each type of memory. Therefore, the next 
+step in maximizing memory throughput is to organize memory accesses as 
+optimally as possible. This is especially important for global memory accesses,
+as global memory bandwidth is low compared to available on-chip bandwidths and
+arithmetic instruction throughput. Thus, non-optimal global memory accesses
+generally have a high impact on performance.
+
+Data Transfer
+-------------------------------------------------------------------------------
+applications should aim to minimize data transfers between the host and the 
+device. This can be achieved by moving more computations from the host to the 
+device, even if it means running kernels that do not fully utilize the 
+parallelism for device. Intermediate data structures can be created, used, 
+and discarded in device memory without being mapped or copied to host memory.
+
+Batching small transfers into a single large transfer can improve performance 
+due to the overhead associated with each transfer. On systems with a front-side
+bus, using page-locked host memory can enhance data transfer performance.
+
+When using mapped page-locked memory, there is no need to allocate device 
+memory or explicitly copy data between device and host memory. Data transfers 
+occur implicitly each time the kernel accesses the mapped memory. For optimal
+performance, these memory accesses should be coalesced, similar to global
+memory accesses.
+
+On integrated systems where device and host memory are physically the same,
+any copy operation between host and device memory is unnecessary, and mapped
+page-locked memory should be used instead. Applications can check if a device
+is integrated by querying the integrated device property.
+
+
+Device Memory Access
+-------------------------------------------------------------------------------
+Memory access instructions may be repeated due to the spread of memory
+addresses across warp threads. The impact on throughput varies with memory type
+and is generally reduced when addresses are more scattered, especially in
+global memory.
+
+Device memory is accessed via 32-, 64-, or 128-byte transactions that must be
+naturally aligned. Maximizing memory throughput involves coalescing memory
```

Comment:
I think a short glossary in the beginning can be very valuable. For example, I'm not sure here, is "device memory" and "global memory" the same thing in these sentences, or it is different concepts.

Reply:
Reference added to coalescing.